code/datums/components/crafting/recipes.dm

```diff
@@ -1326,5 +1326,363 @@
 	time = 40 SECONDS
 	category = CAT_MISC
 
+/datum/crafting_recipe/pipe
+	name = "Smart pipe fitting"
+	tool_behaviors = list(TOOL_WRENCH)
+	result = /obj/item/pipe/quaternary
+	reqs = list(/obj/item/stack/sheet/iron = 1)
+	time = 0.5 SECONDS
+	category = CAT_ATMOSPHERIC
+
+/datum/crafting_recipe/pipe/on_craft_completion(mob/user, atom/result)
+	var/obj/item/pipe/crafted_pipe = result
+	crafted_pipe.pipe_type = /obj/machinery/atmospherics/pipe/smart
+	crafted_pipe.pipe_color = COLOR_VERY_LIGHT_GRAY
+	crafted_pipe.p_init_dir = ALL_CARDINALS
+	crafted_pipe.setDir(SOUTH)
```

Comment:
would it be possible to make this match the direction the crafter is facing?

Reply:
there is no need for that for smart pipes, but i'll do that for the other components